plugins/rebalance/rest.go

```diff
@@ -0,0 +1,231 @@
+	}
+	if Status != STARTED {
+		fmt.Println("Couldn't update status %s", Status)
+		return nil
+	}
+	//start to migrate files, check for failures
+	return nil
+}
+
+func startUndoRebalance(c transaction.TxnCtx) error {
+	var volname string
+	if err := c.Get("volname", &volname); err != nil {
+		c.Logger().WithFields(log.Fields{
+			"error": err,
+			"key":   "volname",
+		}).Error("failed to get value for key from context")
+		return err
+	}
+	var rebalance_id int
+	if err := c.Get("rebalance_id", &rebalance_id); err != nil {
+		c.Logger().WithFields(log.Fields{
+			"error": err,
+			"key":   "rebalance_id",
+		}).Error("failed to get value for key from context")
+		return err
+	}
+	//no. of files started to migrate, restore
+	//need to implement
+	return nil
+}
+
+func registerRebalanceStartStepFuncs() {
+	transaction.RegisterStepFunc(startRebalance, "rebal-start.Commit")
+	transaction.RegisterStepFunc(startUndoRebalance, "rebal-start.Undo")
+}
+
+func rebalanceStart(w http.ResponseWriter, r *http.Request) {
+
+	ctx := r.Context()
+	logger := restutils.GetReqLogger(ctx)
+	//collect inputs from url
+	volname := mux.Vars(r)["volname"]
+	req := new(api.RebStartReq)
+
+	//Check volname given
+	if volname == "" {
+		restutils.SendHTTPError(r.Context(), w, http.StatusBadRequest, "Volume not found", api.ErrCodeDefault)
+		return
+	}
+
+	//Validate rebalance command
+	vol, err := volume.GetVolume(volname)
+	if err != nil {
+		restutils.SendHTTPError(r.Context(), w, http.StatusNotFound, "Invalid volume", api.ErrCodeDefault)
+		return
+	}
+	if vol.State != volume.VolStarted {
+		restutils.SendHTTPError(r.Context(), w, http.StatusBadRequest, "Volume not started", api.ErrCodeDefault)
+		return
+	}
+
+	if vol.DistCount <= 1 {
+		restutils.SendHTTPError(r.Context(), w, http.StatusBadRequest, "Volume is not distribute type or contain only 1 brick",
+			api.ErrCodeDefault)
+		return
+	}
+
+	/*Check volume is a tier volume or not
+	  if vol.Type == volume.Tier {
+	          restutils.SendHTTPError(r.Context(), w, http.StatusBadRequest, "Rebalance operations are not supported on a tired volume",
+	                                  api.ErrCodeDefault)
+	          return
+	  }*/
+
+	//A simple transaction to start rebalance
+	txn := transaction.NewTxn(ctx)
+	defer txn.Cleanup()
+	lock, unlock, err := transaction.CreateLockSteps(volname)
+	if err != nil {
+		restutils.SendHTTPError(r.Context(), w, http.StatusInternalServerError, err.Error(), api.ErrCodeDefault)
+		return
+	}
+	txn.Nodes = vol.Nodes()
+	txn.Steps = []*transaction.Step{
+		lock,
+		{
+			DoFunc:   "rebal-start.Commit",
+			UndoFunc: "rebal-start.Undo",
+			Nodes:    []uuid.UUID{gdctx.MyUUID},
+		},
+		unlock,
+	}
+
+	err = txn.Ctx.Set("volname", volname)
+	if err != nil {
+		logger.WithError(err).Error("failed to set request in transaction context")
+		restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, err.Error(), api.ErrCodeDefault)
+		return
+	}
+
+	reb, err := createRebinfo(req)
```

Comment:
As @kshlm pointed earlier shortening names like this is not corect. Better use `createRebalanceInfo`